### PR TITLE
Prepare 8.8.1 changelog

### PR DESCRIPTION
**Pull request by pchila**
## What does this PR do?

Add changelog for 8.8.1 release

## Checklist

- ~~[ ] My code follows the style guidelines of this project~~
- ~~[ ] I have commented my code, particularly in hard-to-understand areas~~
- ~~[ ] I have made corresponding changes to the documentation~~
- ~~[ ] I have made corresponding change to the default configuration files~~
- ~~[ ] I have added tests that prove my fix is effective or that my feature works~~
- ~~[ ] I have added an entry in `./changelog/fragments` using the [changelog tool](https://github.com/elastic/elastic-agent#changelog)~~
- ~~[ ] I have added an integration test or an E2E test~~

## Author's Checklist

- [ ]

## Related issues

- 

## Questions to ask yourself

- How are we going to support this in production? 
- How are we going to measure its adoption? 
- How are we going to debug this?
- What are the metrics I should take care of?
- ...

### CODE REVIEW

**Comment by pierrehilbert** [comment]
@kilfoyle description is missing in the asciidoc, could you add this manually in the final one on your side?
@endorama do you know why we don't have the description here?

File: changelog/8.8.1.yaml [hunk 1]
```diff
@@ -0,0 +1,28 @@
+version: 8.8.1
+entries:
+    - kind: bug-fix
+      summary: Fix deadlock condition in RuntimeManager
+      description: |
+        In some cases it was possible that a running component would lose connection to the Elastic Agent and
+        unable to re-connect because of a concurrent updated component model. With it being unable to re-connect
+        and the Elastic Agent unable to send it latest component model it results in the Elastic Agent hitting
+        a deadlock where it cannot run properly.
+      component: ""
```
Comment (pierrehilbert) [inline changelog/8.8.1.yaml:10]:
```suggestion
      component: "agent"
```

File: changelog/8.8.1.asciidoc [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+
+
+[discrete]
+[[bug-fixes-8.8.1]]
+=== Bug fixes
+
+
+
+
+* Fix Deadlock Condition In Runtimemanager. {-pull}https://github.com/elastic/elastic-agent/pull/2729[#https://github.com/elastic/elastic-agent/pull/2729] {-issue}https://github.com/elastic/elastic-agent/issues/2691[#https://github.com/elastic/elastic-agent/issues/2691]
```
Comment (kilfoyle) [inline changelog/8.8.1.asciidoc:28]:
```suggestion
* Fix potential deadlock issue in communication between updated components and RuntimeManager. {-pull}https://github.com/elastic/elastic-agent/pull/2729[#https://github.com/elastic/elastic-agent/pull/2729] {-issue}https://github.com/elastic/elastic-agent/issues/2691[#https://github.com/elastic/elastic-agent/issues/2691]
```

File: changelog/8.8.1.asciidoc [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+agent::
+
+* Retry Download Step In Upgrade Process. {agent-pull}https://github.com/elastic/elastic-agent/pull/2776[#https://github.com/elastic/elastic-agent/pull/2776]
```
Comment (kilfoyle) [inline changelog/8.8.1.asciidoc:31]:
```suggestion
* Add a retry download step to the upgrade process. {agent-pull}https://github.com/elastic/elastic-agent/pull/2776[#https://github.com/elastic/elastic-agent/pull/2776] 
```

**Pull request merged**